### PR TITLE
Fixes native button ripple on Android

### PR DESCRIPTION
**Pull request by burkeshartsis**
### What does this PR do, or why is it needed?
This fixes the native button ripple effect on Android when a `Touchable` has rounded corners (e.g. every button in the app!).

This fix just wraps the `TouchableNativeFeedback` that is rendered on Android with a `View` and exposes a `borderRadius` prop on `Touchable` to round the corners of the wrapping `View`. This fix doesn't just fix our buttons though. Any `Icon` that is wrapped in a `Touchable` can now elegantly render Android's ripple effect 🕺

This also sets the Android ripple color based on the Theme `type` (light/dark) so that the color is always appropriate to background 🕺🕺 

#### After
nov-09-2018 11-41-21

#### Theme driven responsive ripple color!
nov-09-2018 12-45-23

### What design trade-offs/decisions were made?
NONE! 💪

### How do I test this PR?
Storybook has examples of buttons. You can also in app for places known to use `Touchable` or `Button`

### What automated tests did you write?
Coverage apparently didn't drop so I didn't add any but there are totally new untested props.

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Partial fix for #382 
- [ ] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apolloschurchapp/src/ui/Touchable/index.js [hunk 1]
```diff
@@ -1,16 +1,45 @@
...
+
+const Touchable = ({ borderRadius, children, ...otherProps }) => (
+  <AndroidWrapper borderRadius={borderRadius}>
+    <PlatformTouchable {...otherProps}>{children}</PlatformTouchable>
+  </AndroidWrapper>
+);
+
+Touchable.propTypes = {
+  borderRadius: PropTypes.number,
+  children: PropTypes.node,
```
Comment (conrad-vanl) [inline packages/apolloschurchapp/src/ui/Touchable/index.js]: This is a good opportunity to make sure we throw a prop type warning if `children` contains multiple elements, since that is invalid on android. I believe `PropTypes.node` will...maybe we should leave a comment here about how on android, `children` most be a single element?

Comment (burkeshartsis) [reply packages/apolloschurchapp/src/ui/Touchable/index.js]: Good idea. https://reactjs.org/docs/typechecking-with-proptypes.html#requiring-single-child

Comment (burkeshartsis) [reply packages/apolloschurchapp/src/ui/Touchable/index.js]: fixed in 663f3f6

**Comment by burkeshartsis** [comment]
@conrad-vanl the media player buttons may not look/function like the gif above. I have another PR coming to make that work correctly. That just shows that it's possible 😉
